### PR TITLE
LibJS: More Temporal editorial changes

### PR DESCRIPTION
**Pull request by linusg**
Two fixes for spec bugs still outstanding. Knowingly regresses one test262 test, will be fixed with > 100 others in the next PR. :^)

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainDate.cpp [hunk 1]
```diff
@@ -486,4 +486,67 @@ i8 compare_iso_date(i32 year1, u8 month1, u8 day1, i32 year2, u8 month2, u8 day2
...
+    if (operation == DifferenceOperation::Since) {
+        // a. Set roundingMode to ! NegateTemporalRoundingMode(roundingMode).
+        rounding_mode = negate_temporal_rounding_mode(rounding_mode);
+    }
+
+    // 12. Let roundingIncrement be ? ToTemporalRoundingIncrement(options, undefined, false).
+    auto rounding_increment = TRY(to_temporal_rounding_increment(global_object, *options, {}, false));
+
+    // 13. Let untilOptions be ? MergeLargestUnitOption(options, largestUnit).
+    auto* until_options = TRY(merge_largest_unit_option(global_object, options, move(*largest_unit)));
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/Temporal/PlainDate.cpp:535]: `release_value()`?

**Review by IdanHo — COMMENTED** [review]
a couple of nitpicks and questions

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp [hunk 2]
```diff
@@ -255,4 +257,92 @@ ThrowCompletionOr<String> temporal_year_month_to_string(GlobalObject& global_obj
...
+{
+    auto& vm = global_object.vm();
+
+    // 1. Let duration be ? ToTemporalDurationRecord(temporalDurationLike).
+    auto duration = TRY(to_temporal_duration_record(global_object, temporal_duration_like));
+
+    // 2. If operation is subtract, then
+    if (operation == ArithmeticOperation::Subtract) {
+        // a. Set duration to ! CreateNegatedTemporalDuration(duration).
+        // FIXME: According to the spec CreateNegatedTemporalDuration takes a Temporal.Duration object,
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp:355]: Nope, just copied over from the old AO. I've filed so many spec issues over the past few days, this one will have to wait (or someone else does it) :^)

File: Userland/Libraries/LibJS/Runtime/Temporal/Instant.cpp [hunk 3]
```diff
@@ -119,7 +119,7 @@ ThrowCompletionOr<BigInt*> parse_temporal_instant(GlobalObject& global_object, S
     auto* utc = get_epoch_from_iso_parts(global_object, result.year, result.month, result.day, result.hour, result.minute, result.second, result.millisecond, result.microsecond, result.nanosecond);
 
     // 6. If ℝ(utc) < -8.64 × 10^21 or ℝ(utc) > 8.64 × 10^21, then
-    if (utc->big_integer() < INSTANT_NANOSECONDS_MIN || utc->big_integer() > INSTANT_NANOSECONDS_MAX) {
+    if (utc->big_integer() < "-8640000000000000000000"_sbigint || utc->big_integer() > "8640000000000000000000"_sbigint) {
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/Instant.cpp]: [Left a comment](https://github.com/tc39/proposal-temporal/pull/2182#pullrequestreview-965364036), updated in code :+1:

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp [hunk 4]
```diff
@@ -255,4 +257,92 @@ ThrowCompletionOr<String> temporal_year_month_to_string(GlobalObject& global_obj
...
+        //        not a record, so we have to do some trickery. If they want to accept anything that has
+        //        the required internal slots, this should be updated in the AO's description.
+        //        We also have to convert back to a Duration Record afterwards to match the initial type.
+        auto* actual_duration = MUST(create_temporal_duration(global_object, duration.years, duration.months, duration.weeks, duration.days, duration.hours, duration.minutes, duration.seconds, duration.milliseconds, duration.microseconds, duration.nanoseconds));
+        auto* negated_duration = create_negated_temporal_duration(global_object, *actual_duration);
+        duration = TRY(to_temporal_duration_record(global_object, negated_duration));
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/PlainYearMonth.cpp]: Yep, copy & paste mistake from the old AO, I think - fixed

File: Userland/Libraries/LibJS/Runtime/Temporal/PlainDateTime.cpp [hunk 5]
```diff
@@ -68,25 +68,20 @@ BigInt* get_epoch_from_iso_parts(GlobalObject& global_object, i32 year, u8 month
...
 // 5.5.2 ISODateTimeWithinLimits ( year, month, day, hour, minute, second, millisecond, microsecond, nanosecond ), https://tc39.es/proposal-temporal/#sec-temporal-isodatetimewithinlimits
 bool iso_date_time_within_limits(GlobalObject& global_object, i32 year, u8 month, u8 day, u8 hour, u8 minute, u8 second, u16 millisecond, u16 microsecond, u16 nanosecond)
 {
     // 1. Let ns be ℝ(GetEpochFromISOParts(year, month, day, hour, minute, second, millisecond, microsecond, nanosecond)).
     auto ns = get_epoch_from_iso_parts(global_object, year, month, day, hour, minute, second, millisecond, microsecond, nanosecond)->big_integer();
 
-    // 2. If ns ≤ -8.64 × 10^21 - 8.64 × 10^13, then
-    if (ns <= DATETIME_NANOSECONDS_MIN) {
+    // 2. If ns ≤ nsMinInstant - nsPerDay, then
+    if (ns <= ns_min_instant.minus(Crypto::SignedBigInteger::create_from((i64)ns_per_day))) {
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/PlainDateTime.cpp]: Reverted to old constants

File: Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp [hunk 6]
```diff
@@ -588,11 +588,11 @@ ThrowCompletionOr<Instant*> disambiguate_possible_instants(GlobalObject& global_
     // 7. Let epochNanoseconds be GetEpochFromISOParts(dateTime.[[ISOYear]], dateTime.[[ISOMonth]], dateTime.[[ISODay]], dateTime.[[ISOHour]], dateTime.[[ISOMinute]], dateTime.[[ISOSecond]], dateTime.[[ISOMillisecond]], dateTime.[[ISOMicrosecond]], dateTime.[[ISONanosecond]]).
     auto* epoch_nanoseconds = get_epoch_from_iso_parts(global_object, date_time.iso_year(), date_time.iso_month(), date_time.iso_day(), date_time.iso_hour(), date_time.iso_minute(), date_time.iso_second(), date_time.iso_millisecond(), date_time.iso_microsecond(), date_time.iso_nanosecond());
 
-    // 8. Let dayBefore be ! CreateTemporalInstant(epochNanoseconds - 8.64 × 10^13ℤ).
-    auto* day_before = MUST(create_temporal_instant(global_object, *js_bigint(vm, epoch_nanoseconds->big_integer().minus("86400000000000"_sbigint))));
+    // 8. Let dayBefore be ! CreateTemporalInstant(epochNanoseconds - ℤ(nsPerDay)).
+    auto* day_before = MUST(create_temporal_instant(global_object, *js_bigint(vm, epoch_nanoseconds->big_integer().minus(Crypto::SignedBigInteger::create_from((i64)ns_per_day)))));
```
[the remark this answers is not in the document]
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/Temporal/TimeZone.cpp]: Replaced RHS with a constant, LHS is different every time and cannot be optimized

**Pull request merged**